firestore/src/main/java/com/firebase/ui/firestore/FirestoreArray.java

```diff
@@ -110,13 +110,16 @@ private void onDocumentModified(DocumentChange change) {
             // Document modified only
             mSnapshots.set(change.getNewIndex(), change.getDocument());
             notifyOnChildChanged(ChangeEventType.CHANGED, change.getDocument(),
-                    change.getNewIndex(), change.getOldIndex());
+                    change.getNewIndex(), change.getNewIndex());
         } else {
             // Document moved and possibly also modified
             mSnapshots.remove(change.getOldIndex());
             mSnapshots.add(change.getNewIndex(), change.getDocument());
+
             notifyOnChildChanged(ChangeEventType.MOVED, change.getDocument(),
                     change.getNewIndex(), change.getOldIndex());
+            notifyOnChildChanged(ChangeEventType.CHANGED, change.getDocument(),
+                    change.getNewIndex(), change.getNewIndex());
```

Comment:
Should we try to no-op this? I can't think of a use case where a change isn't caused by the model also changing...